x-pack/plugins/synthetics/public/legacy_uptime/state/private_locations/api.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObjectsClientContract } from '@kbn/core/public';
+import { AgentPoliciesList } from '.';
+import {
+  privateLocationsSavedObjectId,
+  privateLocationsSavedObjectName,
+} from '../../../../common/saved_objects/private_locations';
+import { apiService } from '../api/utils';
+import { SyntheticsPrivateLocations } from '../../../../common/runtime_types';
+
+const FLEET_URLS = {
+  AGENT_POLICIES: '/api/fleet/agent_policies',
+};
+
+export const fetchAgentPolicies = async (): Promise<AgentPoliciesList> => {
+  return await apiService.get(
+    FLEET_URLS.AGENT_POLICIES,
+    { page: 1, perPage: 10000, sortField: 'name', sortOrder: 'asc', full: true },
```

Comment:
I sincerely hope no one is managing a fleet of 1000 agents, but if they are, in the future we may need to have a search feature so that users can find the specific agent policy to attach a location to.